File: problems/prob1/2004.cpp
#include <iostream>
#include <cmath>

using namespace std;

int getNum(int a, int b) {
    int cnt = 0;
    for(unsigned long long i = a; i <= b; i *= a) {
        cnt += b / i;
    }
    return cnt;
}

int main() {

    unsigned long long n, m;
    int start;
    int cnt_2 = 0;
    int cnt_5 = 0;

    cin >> n >> m;
    
    cnt_2 += getNum(2, n);
    cnt_2 -= getNum(2, m);
    cnt_2 -= getNum(2, n-m);

    cnt_5 += getNum(5, n);
    cnt_5 -= getNum(5, m);
    cnt_5 -= getNum(5, n-m);

    cout << min(cnt_2, cnt_5) << '\n';

    return 0;
}

File: problems/prob1/2004.py
n, m = map(int, input().split())

cnt_2 = 0
cnt_5 = 0

start = 0

# Count 5
for i in range(5):
    if (n-i)%5 == 0:
        start = n-i
        break

for i in range(start, n-m, -5):
    tmp = i
    while tmp%5 == 0:
        cnt_5 += 1
        tmp /= 5

for i in range(5, m-1, 5):
    tmp = i
    while tmp%5 == 0:
        cnt_5 -= 1
        tmp /= 5

# Count 2
if n%2 == 0:
    start = n
else:
    start = n-1

for i in range(start, n-m, -2):
    tmp = i
    while tmp%2 == 0:
        cnt_2 += 1
        tmp /= 2

for i in range(2, m+1, 2):
    tmp = i
    while tmp%2 == 0:
        cnt_2 -= 1
        tmp /= 2

print(min(cnt_2, cnt_5))